tensorflow/python/keras/callbacks.py

```diff
@@ -1299,20 +1301,38 @@ def _save_model(self, epoch, logs):
                         'ModelCheckpoint. Filepath used is an existing '
                         'directory: {}'.format(filepath))
 
-  def _get_file_path(self, epoch, logs):
+  def _get_file_path(self, epoch, batch, logs):
     """Returns the file path for checkpoint."""
     # pylint: disable=protected-access
-    try:
-      # `filepath` may contain placeholders such as `{epoch:02d}` and
-      # `{mape:.2f}`. A mismatch between logged metrics and the path's
-      # placeholders can cause formatting to fail.
-      file_path = self.filepath.format(epoch=epoch + 1, **logs)
-    except KeyError as e:
-      raise KeyError('Failed to format this callback filepath: "{}". '
-                     'Reason: {}'.format(self.filepath, e))
-    self._write_filepath = distributed_file_utils.write_filepath(
+    if not self.model._in_multi_worker_mode(
```

Comment:
We shouldn't need this if-else anymore since distributed_file_utils.write_filepath would take care of that writing path. Can you fix this?

Reply:
Done